File: singly_linked_list/recipe-02/c/test_delete_tail.c
#include "string_list.h"
#include <stdio.h>

void print(ItemType data) 
{
    printf("data: %s\n", data);
}

void test()
{
    SinglyLinkedList* words = NewList();
    ListAppend(words, "foo");
    ListAppend(words, "bar");
    ListAppend(words, "bim");
    ListAppend(words, "baz");
    ListAppend(words, "quux");

    printf("The origin list\n");
    printf("size: %d\n", ListCount(words));
    ListTravel(words, &print);

    printf("Delete the last item in the list\n");
    ListDelete(words, "quux");
    printf("size: %d\n", ListCount(words));
    ListTravel(words, &print);

    printf("append another item\n");
    ListAppend(words, "Quux");
    printf("size: %d\n", ListCount(words));
    ListTravel(words, &print);

    FreeList(words);
}

int main()
{
    Init();

    test();

    return 0;
}

File: list/singly_linked_list/recipe-01/python/test_delete_tail.py
from singly_linked_list import SinglyLinkedList

def test():
    words = SinglyLinkedList()
    words.Append('foo')
    words.Append('bar')
    words.Append('bim')
    words.Append('baz')
    words.Append('quux')
    
    print("The origin list");
    print("size: {}".format(words.count))
    for word in words:
        print("data: {}".format(word))

    print("Delete the last item in the list")
    words.Delete('quux')
    print("size: {}".format(words.count))
    for word in words:
        print("data: {}".format(word))

    print("append another item")
    words.Append('Quux')
    print("size: {}".format(words.count))
    for word in words:
        print("data: {}".format(word))

if __name__ == "__main__":
    test()
